lib/activerecord_sqlserver_adapter/jdbc/dataset.rb

```diff
@@ -0,0 +1,225 @@
+module ActiveRecordSqlServerAdapter
+  module Jdbc
+    class Dataset
+      include Enumerable
+
+      OPTS = {}.freeze
+
+      def synchronize(&block)
+        @cache_mutex.synchronize(&block)
+      end
```

Comment:
I have concerns on adding stuff like this and why it is needed so I would first like to know the full details on what this is used for? Is the JDBC connection not thread safe? I though most of Rails v5.1 and upward are leveraging [concurrent-ruby](https://github.com/ruby-concurrency/concurrent-ruby) now too? But first, I'd just like to know more about this.